07.ls_object/file_detail.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class FileDetail
+  TYPE_LIST =
+    {
+      file: '-',
+      directory: 'd',
+      characterSpecial: 'c',
+      blockSpecial: 'b',
+      fifo: 'p',
+      link: 'l',
+      socket: 's'
+    }.freeze
+
+  MODE_LIST =
+    {
+      7 => 'rwx',
+      6 => 'rw-',
+      5 => 'r-x',
+      4 => 'r--',
+      3 => '-wx',
+      2 => '-w-',
+      1 => '--x',
+      0 => '---'
+    }.freeze
+
+  attr_reader :name, :stat
+
+  def initialize(name, directory = '')
+    @name = name
+    path = if directory == ''
```

Comment:
```suggestion
    path = if directory.empty?
```

みたいに書くとRubyらしいかも（好みの問題）